src/serializers/infer.rs

```diff
@@ -117,6 +117,7 @@ pub(crate) fn infer_to_python_known(
         serializer.serializer.to_python(value, include, exclude, &extra)
     };
 
+    // TODO: add support for inf_nan_mode
```

Comment:
Last thing to do here, I'm a bit stuck on this @davidhewitt

Reply:
I think you need to change the `ObType::Float` and `ObType::FloatSubclass` branches, so that if the value would be non-finite rather than calling `value.into_py(py)` you instead return `None` (according to the config).

Reply:
Ok, I attempted to do this 👍